Add python_restart_enable server option

toonen opened the pull request:
This PR adds a new server option, `python_restart_enable`, that allows the Python interpreter restart to be enabled or disabled.  The default value is enabled.  This PR is related to PR #2501.

bayucan commented on src/lib/Libpython/pbs_python_svr_internal.c:
```diff
@@ -5252,14 +5255,27 @@ _pbs_python_event_set(unsigned int hook_event, char *req_user, char *req_host,
+		enable_restart = PBS_PYTHON_RESTART_ENABLE;
+	if (lval != enable_restart) {
+		snprintf(log_buffer, sizeof(log_buffer),
+			"python_disable_restart is now %ld", enable_restart);
```
The log message talks about "python_disable_restart" but the attribute is "python_enable_restart".

toonen replied:
Apparently I missed one when I renamed the attribute to a more positive name.  Fixed in 97498d40fc6d249c395d7714dfc15ca99b53455a.

BTW, if the attribute is unset, line 5262 sets the default to enabling the interpreter restart to maintain backwards compatibility.  Do I need to set the default somewhere else as well?

bayucan replied:
@toonen : ah I see, it's okay then.

bayucan reviewed (changes requested):
- This needs a PTL test case.
- Also, there should be a design doc since this adds a new 'python_restart_enable' server attribute.
- I think this server attribute should have a value of 1 if unset, to maintain backwards compatibility behavior when restarting python interpreter. So that must also be coded.

vishwaks commented on src/lib/Libpython/pbs_python_svr_internal.c:
```diff
@@ -5252,14 +5255,27 @@ _pbs_python_event_set(unsigned int hook_event, char *req_user, char *req_host,
+		enable_restart = PBS_PYTHON_RESTART_ENABLE;
+	if (lval != enable_restart) {
+		snprintf(log_buffer, sizeof(log_buffer),
+			"python_enable_restart is now %ld", enable_restart);
```
log correct option name "python_restart_enable" .

bhroam replied:
There is a logging function call log_eventf() which combines both sprintf() and log_event().  Consider using it instead of creating a string to just print.

toonen commented:
Given our current priorities, and that this is not a critical feature, I am closing this PR until we can address the reviewers requests and comments.  We will open a new PR once we have addressed the issues documented in this PR.
